drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1670 @@
+  local op_type = DoorLock.types.OperatingModeEnum
+  local opMode_map = {
+    [op_type.NORMAL] = true,
+    [op_type.VACATION] = true,
+    [op_type.PRIVACY] = false,
+    [op_type.NO_REMOTE_LOCK_UNLOCK] = false,
+    [op_type.PASSAGE] = false,
+  }
+  local result = opMode_map[ib.data.value]
+  if result == true then
+    device:emit_event(status("true", {visibility = {displayed = true}}))
+    device:emit_event(capabilities.lock.supportedLockCommands({"lock", "unlock"}, {visibility = {displayed = false}}))
+  elseif result == false then
+    device:emit_event(status("false", {visibility = {displayed = true}}))
+    device:emit_event(capabilities.lock.supportedLockCommands({}, {visibility = {displayed = false}}))
+  end
+end
+-------------------------------------
+-- Number Of Total Users Supported --
+-------------------------------------
+local function total_users_supported_handler(driver, device, ib, response)
+  device:emit_event(capabilities.lockUsers.totalUsersSupported(ib.data.value, {visibility = {displayed = false}}))
+end
+
+----------------------------------
+-- Number Of PIN Users Supported --
+----------------------------------
+local function pin_users_supported_handler(driver, device, ib, response)
+  device:emit_event(capabilities.lockCredentials.pinUsersSupported(ib.data.value, {visibility = {displayed = false}}))
+end
+
+-------------------------
+-- Min PIN Code Length --
+-------------------------
+local function min_pin_code_len_handler(driver, device, ib, response)
+  device:emit_event(capabilities.lockCredentials.minPinCodeLen(ib.data.value, {visibility = {displayed = false}}))
+end
+
+-------------------------
+-- Max PIN Code Length --
+-------------------------
+local function max_pin_code_len_handler(driver, device, ib, response)
+  device:emit_event(capabilities.lockCredentials.maxPinCodeLen(ib.data.value, {visibility = {displayed = false}}))
+end
+
+--------------------------------------
+-- Require PIN For Remote Operation --
+--------------------------------------
+--- If a device needs a cota credential this function attempts to set the credential
+--- at the index provided. The set_credential_response_handler handles all failures
+--- and retries with the appropriate index when necessary.
+local function set_cota_credential(device, credential_index)
+  local eps = device:get_endpoints(DoorLock.ID)
+  local cota_cred = device:get_field(lock_utils.COTA_CRED)
+  if cota_cred == nil then
+    -- Shouldn't happen but defensive to try to figure out if we need the cota cred and set it.
+    device:send(DoorLock.attributes.RequirePINforRemoteOperation:read(device, #eps > 0 and eps[1] or 1))
+    return
+  elseif cota_cred == false then
+    device.log.debug("Device does not require PIN for remote operation. Not setting COTA credential")
+    return
+  end
+
+  if device:get_field(lock_utils.SET_CREDENTIAL) ~= nil then
+    device.log.debug("delaying setting COTA credential since a credential is currently being set")
```

Comment:
When I test with a lock that uses COTA and the initial attempt to set the COTA PIN fails with a "duplicate" error, then it gets stuck in a loop logging this message and failing to ever retry setting the COTA PIN.